Three inflections posts for March 2026 (#245)
andycroll/andycroll.com: _posts/2026/2026-03-16-declare-acronyms-in-rails-inflections.md

## Instead of...

...accepting the wrong casing in your class names:

```ruby
# app/controllers/api/users_controller.rb
class Api::UsersController < ApplicationController
end
```

## Use...

...`inflect.acronym` to teach Rails the correct casing:

```ruby
# config/initializers/inflections.rb
ActiveSupport::Inflector.inflections(:en) do |inflect|
  inflect.acronym "API"
end
```

Now Rails expects `API::UsersController`. The file path stays lowercase (`app/controllers/api/`), but the class name uses the acronym:

```ruby
# app/controllers/api/users_controller.rb
class API::UsersController < ApplicationController
end
```

## Why?

The `acronym` method tells ActiveSupport's inflector to preserve the casing you specify. It affects `camelize`, `underscore`, `classify`, and `titleize`—which means it also affects autoloading and URL helpers.

```ruby
"api".camelize        #=> "API"
"API".underscore      #=> "api"
"api/users".camelize  #=> "API::Users"
```

Without the acronym declaration, you get `Api` instead:

```ruby
"api".camelize  #=> "Api"
```

> Unlike irregular plurals and uncountable words, Rails ships with [no built-in acronyms](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/inflections.rb)—every one you need, you have to declare yourself. Common ones worth adding: `API`, `SMS`, `CSV`, `HTML`, `PDF`. One call per term:

Reviewer: ```suggestion
Unlike irregular plurals and uncountable words, Rails ships with [no built-in acronyms](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/inflections.rb)—every one you need, you have to declare yourself. Common ones worth adding: `API`, `SMS`, `CSV`, `HTML`, `PDF`. You need one call per term:
```

```ruby
ActiveSupport::Inflector.inflections(:en) do |inflect|
  inflect.acronym "API"
  inflect.acronym "SMS"
  inflect.acronym "CSV"
end
```

This also works for mixed-case words like `GraphQL` or `GitHub`. `inflect.acronym "GraphQL"` ensures `"graphql".camelize` returns `"GraphQL"` rather than `"Graphql"`.

See the [ActiveSupport::Inflector::Inflections documentation](https://api.rubyonrails.org/classes/ActiveSupport/Inflector/Inflections.html#method-i-acronym) for details.

> 

Author: ```suggestion

Note that because these changes are in an initializer, you'll need to restart your Rails server after making changes.

```

## Why not?

> Since it's an initialiser, you'll need to restart your Rails server after making changes.

Reviewer: Not really a reason not to use it if you need it, right?

Author: True.

> 

Author: ```suggestion
```

> Keep the list short. Every entry changes how `camelize`, `titleize`, `humanize`, and `underscore` behave across your entire app. Only add acronyms you're actively using—whether in class names, attribute labels, or view helpers.

Reviewer: *behave _for the specified words/entries_ ?

Reviewer: But another super strong post! 😍

Author: ```suggestion
Keep the list short. Every entry changes how `camelize`, `titleize`, `humanize`, and `underscore` behave for the specified words across your entire app. Only add acronyms you're actively using—whether in class names, attribute labels, or view helpers.
```

> This only affects casing, not pluralisation. For words with non-standard plurals like criterion/criteria, that's `inflect.irregular`. For words that don't pluralise at all, that's `inflect.uncountable`.

Reviewer: ```suggestion
This only affects casing, not pluralisation. For words with non-standard plurals like criterion/criteria, you'll want `inflect.irregular`. For words that don't pluralise at all, the method to look at is `inflect.uncountable`.
```
